sympy/core/numbers.py

```diff
@@ -1374,7 +1375,7 @@ def __int__(self):
         if self._mpf_ == fzero:
             return 0
         return int(mlib.to_int(self._mpf_))  # uses round_fast = round_down
-
+#ping
     def __eq__(self, other):
```

Comment:
It would be nice if this only uses `_sympify` on appropriate classes, rather than any, but which are those?

It might be better to replace `isinstance(other, Boolean)` with `other.is_Bool` to avoid the circular import.

Reply:
I think it is prefered to use `return NotImplemented` over `return other.__eq__(self)` and let python try the other class.